deploy/cdk/marble-web-kiosk-export-pipeline/lib/marble-web-kiosk-export-pipeline-stack.ts

```diff
@@ -0,0 +1,259 @@
+          stageName: 'DeployToProd',
+          actions: [ deployToProdAction ],
+        })
+      }
+
+      setup () {
+        // This must match the stack name of the marble-web-kiosk-export project without the stage suffix.
+        // This is also the same as the GitHub repository name.
+        const localStackName = 'marble-web-kiosk-export'
+
+        const repoName = this.node.tryGetContext('repoName') ||
+          readlineSync.question('What is the name of the GitHub repository that has the application source code? (default: ' + localStackName + ') ', {
+            defaultInput: localStackName,
+          })
+        const repoOwner = this.node.tryGetContext('repoOwner') ||
+          readlineSync.question('What is the owner name for the GitHub repository specified? (default: ndlib) ', {
+            defaultInput: 'ndlib',
+          })
+        const repoBranch = this.node.tryGetContext('repoBranch') ||
+          readlineSync.question('Which branch do you want to build from? (default: master) ', {
+            defaultInput: 'master',
+          })
+
+        // S3 BUCKET FOR STORING ARTIFACTS
+        const artifactBucket = new Bucket(this, 'ArtifactBucket', {
+          removalPolicy: RemovalPolicy.DESTROY,
+        })
+        artifactBucket.addToResourcePolicy(new PolicyStatement({
+          principals: [new AnyPrincipal()],
+          effect: Effect.DENY,
+          actions: ['s3:*'],
+          conditions: {
+            'Bool': { 'aws:SecureTransport': false }
+          },
+          resources: [artifactBucket.bucketArn + '/*']
+        }))
+
+        // SNS TOPICS
+        const approvalTopic = new sns.Topic(this, 'PipelineApprovalTopic', {
+          displayName: 'PipelineApprovalTopic',
+        })
+
+        // IAM ROLES
+        const codepipeline_role = new Role(this, 'CodePipelineRole', {
+          assumedBy: new ServicePrincipal('codepipeline.amazonaws.com')
+        })
+
+        const codebuild_role = new Role(this, 'CodeBuildTrustRole', {
+          assumedBy: new ServicePrincipal('codebuild.amazonaws.com')})
+          // Allow checking what policies are attached to this role and any roles the application stack creates
+          // When cdk makes roles, it removes all punctuation. The application stack should prefix the LogicalId
+          // of any construct that needs roles with the stack name so that it matches the wildcard here.
+          const rolePattern = localStackName.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "") + '*'
+          codebuild_role.addToPolicy(new PolicyStatement({
+            resources: [
+              codebuild_role.roleArn,
+              Fn.sub('arn:aws:iam::${AWS::AccountId}:role/' + rolePattern),
+            ],
+            actions: ['iam:GetRolePolicy'],
+          }))
+
+        // Allow access to Parameter Store
+        codebuild_role.addToPolicy(new PolicyStatement({
+            resources:[Fn.sub('arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:*')],
```

Comment:
Can this be narrowed to the path(s) your project will need?

Reply:
Thanks for the suggestion.  I have just push up a resolution.  Could you please review that change?  Thanks.